packages/commonwealth/client/scripts/state/api/topics/setTopicThreshold.ts

```diff
@@ -0,0 +1,34 @@
+import axios from 'axios';
+import app from 'state';
+import { useMutation } from '@tanstack/react-query';
+import { ApiEndpoints, queryClient } from 'state/api/config';
+import Topic from 'models/Topic';
+
+interface SetTopicThresholdProps {
+  topic: Topic;
+  topicThreshold: string;
+}
+
+const setTopicThreshold = async ({
```

Comment:
can we move this to `editTopic`, (since its also a resource mutation endpoint)? This would help if we refactor the topics endpoint to be more RESTful (http verb centric). 

Looks good for now tho.

Reply:
IMO we ideally should refactor those similarly to threads and create a backend around RESTful pattern. I agree that we should consolidate setTopicThreshold and updateFeaturedTopicsOrder into editTopic but I am not sure that we need to do this right now. Very good point though

Reply:
>  agree that we should consolidate setTopicThreshold and updateFeaturedTopicsOrder into editTopic but I am not sure that we need to do this right now.

I think we are good for now, just wanted to raise this point. Let's do the consolidation in a future update!